File: insertion_sort/algorithm.py

def insertion_sort(A: list, size: int):
    """
    inputs:
        A: list = it is an array of numbers to be sorted 
    """
    # [7, 3, 5, 6, 1, 0, 2]
    for i in range(1, size):
        key = A[i]
        j = i - 1
        while j >= 0 and A[j] > key:
            A[j+1] = A[j]
            j = j - 1
        A[j + 1] = key


def main():
    unsorted_array = [7, 3, 5, 6, 1, 0, 2]
    size = len(unsorted_array)
    insertion_sort(unsorted_array, size)
    return unsorted_array



if __name__ == "__main__":
    print(main())

File: insertion_sort/algorithm.c
#include <stdio.h>

void insertion_sort(int A[], int size) {
    for (int i = 1; i < size; i++)
    {
        int key = A[i];

        int j = i - 1;
        while (j >= 0 && A[j] > key)
        {
           A[j + 1] = A[j];
           j = j - 1; 
        }
        A[j + 1] = key;    
    }
    
    
}

int main() {
    int array[] = {7, 3, 5, 6, 1, 0, 2, 1};
    int size = sizeof(array) / sizeof(array[0]);
    insertion_sort(array, size);

    printf("[");
    for (int i = 0; i < size; i++)
    {
        printf("%d, ", array[i]);
    }
    printf("]\n");
    
    return 0;

}